tornadoAPI.py

```diff
@@ -0,0 +1,421 @@
+        return None
+
+    accounts = (
+        os.environ["TORNADO"].strip().split(",")
+        if TORNADO_EXTERNAL is None
+        else TORNADO_EXTERNAL.strip().split(",")
+    )
+    TORNADO_obj = Brokerage("TORNADO")
+    
+    for index, account in enumerate(accounts):
+        account_name = f"Tornado {index + 1}"
+        try:
+            driver = getDriver(DOCKER=False)
+            if driver is None:
+                raise Exception("Driver not found.")
+            driver.get('https://tornado.com/app/login')
+            WebDriverWait(driver, 30).until(check_if_page_loaded)
+
+            # Log in with email and password
+            try:
+                email_field = WebDriverWait(driver, 30).until(
+                    EC.element_to_be_clickable((By.ID, "email-field")))
+                email_field.send_keys(account.split(":")[0])
+
+                password_field = WebDriverWait(driver, 30).until(
+                    EC.element_to_be_clickable((By.ID, "password-field")))
+                password_field.send_keys(account.split(":")[1])
+
+                login_button = WebDriverWait(driver, 30).until(
+                    EC.element_to_be_clickable((By.CSS_SELECTOR, "#root > div > div > div > div:nth-child(2) > div > div > div > div > form > div.sc-WZYut.ZaYjk > button")))
+                login_button.click()
+
+                # Check for the element after logging in to ensure the page is fully loaded
+                WebDriverWait(driver, 60).until(
+                    EC.presence_of_element_located((By.XPATH, "//*[@id='main-router']/div/div/div/div[1]/div/div/div/div[1]/div[1]/div/span"))
+                )
+
+                TORNADO_obj.set_logged_in_object(account_name, driver)
+
+                # Set the account name
+                TORNADO_obj.set_account_number(account_name, account_name)
+                logger.info(f"Set account name for {account_name}")
+
+            except TimeoutException:
+                printAndDiscord(f"TimeoutException: Login failed for {account_name}.", loop)
+                return False
+
+        except Exception as e:
+            tornado_error(driver, e, loop)
+            driver.close()
+            driver.quit()
+            return None
+    return TORNADO_obj
+
+
+def tornado_extract_holdings(driver):
+    holdings_data = []
+    try:
+        # Locate all the individual stock holdings elements
+        holdings_elements = driver.find_elements(By.XPATH, ".//div[@class='sc-jEWLvH evXkie']")
+
+        logger.info("Found %d holdings elements to process.", len(holdings_elements))
+
+        for holding_element in holdings_elements:
+            try:
+                # Extract the stock ticker
+                stock_ticker = holding_element.find_element(By.XPATH, ".//a[1]/div[1]/span").text.strip()
+
+                # Extract the number of shares available
+                shares = holding_element.find_element(By.XPATH, ".//a[4]/div/div/span/span").text.strip()
```

Comment:
If there is no shares in the account this throws a NoSuchElementException. This situation needs gracefully handled so console doesn't get filled with error messages.

Reply:
You're right. I have lots of stocks I'm unable to sell, so it's impossible for my account, at least not to have shares.

Reply:
![image](https://github.com/user-attachments/assets/42a41142-ba57-4db2-ab18-91d0591d9520)
The bot is still correct when the error is thrown, there is just spam in the console that needs handled if that helps you at all.

Reply:
Hooray, I don't really use the command line interface, but for the people that do, I will make it cleaner. ALSO, you can see that the account number on Tornado holsigns is a little weird, and that's because it's impossible to scrape. The only way I found to get the hidden account number is from the statement pdf's. I can't seem to find it anywhere inside the website. If you can figure it out please let me know.